src/main/java/uk/gov/hmcts/reform/notificationservice/service/NotificationService.java

```diff
@@ -0,0 +1,62 @@
+package uk.gov.hmcts.reform.notificationservice.service;
+
+import org.slf4j.Logger;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.notificationservice.clients.ErrorNotificationClient;
+import uk.gov.hmcts.reform.notificationservice.clients.ErrorNotificationRequest;
+import uk.gov.hmcts.reform.notificationservice.clients.ErrorNotificationResponse;
+import uk.gov.hmcts.reform.notificationservice.data.Notification;
+import uk.gov.hmcts.reform.notificationservice.data.NotificationRepository;
+
+import java.util.List;
+
+import static org.slf4j.LoggerFactory.getLogger;
+
+@Service
+public class NotificationService {
+
+    private static final Logger log = getLogger(NotificationService.class);
+
+    private final NotificationRepository notificationRepository;
+    private final ErrorNotificationClient notificationClient;
+
+    public NotificationService(
+        NotificationRepository notificationRepository,
+        ErrorNotificationClient notificationClient
+    ) {
+        this.notificationRepository = notificationRepository;
+        this.notificationClient = notificationClient;
+    }
+
+    public void processPendingNotifications() {
```

Comment:
should be Transactional I assume

Reply:
that will be for the whole lot of notifications. will it work for `private void processNotifications` method down below?